internal/kldtx/txnprocessor.go

```diff
@@ -349,6 +349,15 @@ func (p *txnProcessor) cancelInFlight(inflight *inflightTxn, gapPotential bool)
 				}
 			}
 		}
+		if higherNonceInflight < 0 {
+			// We did not find a higher nonce in-flight, so there's no gap to fill. However, we need to make sure
+			// that this nonce is re-used by the next incoming transaction for this address, if this is not the first transaction for the address.
+			// If this is the first txn, we would have cleared the key from the map - so there's nothing to update.
+			if gapPotential && p.inflightTxns[inflight.from] != nil {
+				log.Infof("Did not find a Transaction with higher nonce in-flight, setting highest nonce for %s to %d", inflight.from, inflight.nonce-1)
+				p.inflightTxns[inflight.from].highestNonce = inflight.nonce - 1
```

Comment:
Looks good. Reviewed the code above to confirm in the case we were the last transaction in flight for this nonce, we would have cleared the entry from `inflightTxns` already. So the additional nil check on the map catches that.